WooCommerce/Classes/Extensions/String+Helpers.swift

```diff
@@ -68,4 +68,150 @@ extension String {
+
+                let characterAtIndexIsNumberOrDot = characterAtIndex.isNumber || characterAtIndex == "."
+                let characterBeforeIsNumberOrDot = characterBefore.isNumber || characterBefore == "."
+
+                if !characterAtIndexIsNumberOrDot && characterBefore.isNumber {
+                    outString += "."
+                }
+
+                if characterAtIndex.isNumber && !characterBeforeIsNumberOrDot {
+                    outString += "."
+                }
+
+                outString += String(characterAtIndex)
+            }
+
+            return outString
+        }
+
+        /// Ranked per https://www.php.net/manual/en/function.version-compare.php
+        ///
+        enum ComponentScore: Comparable {
+            case other
+            case dev
+            case alpha
+            case beta
+            case RC
+            case number
+            case patch
+        }
+
+        /// Score each component
+        ///
+        func scoreComponent(stringComponent: String) -> ComponentScore {
+            if stringComponent == "dev" {
+                return .dev
+            }
+            if stringComponent == "alpha" || stringComponent == "a" {
+                return .alpha
+            }
+            if stringComponent == "beta" || stringComponent == "b" {
+                return .beta
+            }
+            if stringComponent == "RC" || stringComponent == "rc" {
+                return .RC
+            }
+            let componentCharacterSet = CharacterSet(charactersIn: stringComponent)
+            if componentCharacterSet.isSubset(of: .decimalDigits) {
+                return .number
+            }
+
+            if stringComponent == "pl" || stringComponent == "p" {
+                return .patch
+            }
+
+            return .other
+        }
+
+        /// Score and compare two string components
+        ///
+        func compareStringComponent(stringComponent1: String, stringComponent2: String) -> ComparisonResult {
+            /// Score each component
+            let stringComponent1Score = scoreComponent(stringComponent: stringComponent1)
+            let stringComponent2Score = scoreComponent(stringComponent: stringComponent2)
+
+            if stringComponent1Score < stringComponent2Score {
+                return .orderedAscending
+            }
+
+            if stringComponent1Score > stringComponent2Score {
+                return .orderedDescending
+            }
+
+            if stringComponent1Score == .number && stringComponent2Score == .number {
+                let component1Int = Int(stringComponent1) ?? 0
+                let component2Int = Int(stringComponent2) ?? 0
+
+                if component1Int < component2Int {
+                    return .orderedAscending
+                }
+
+                if component1Int > component2Int {
+                    return .orderedDescending
+                }
```

Comment:
nit: maybe can simplify this with `NSNumber(component1Int).compare(component2Int)`?

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/5382/commits/a81193104a67fd74b4aa14d2db66db85f040b764